### ISSUE TITLE
no-fallthrough failing on continue statement

### ISSUE THREAD

**Issue by devangnegandhi** [eslint/eslint#1220]
Take a look at the following code

``` javascript
var i;
for (i = 0; i < 1; i += 1) {
    switch (i) {
    case 1:
        continue;
    default:
        break;
    }
}
```

When I run eslint with the `no-fallthrough` rule turned on, I get the following error

``` bash
test.js
  4:4  error  Expected a "break" statement before "default"  no-fallthrough
```

To fix the issue, I added a `break;` statement as recommended

``` javascript
var i;
for (i = 0; i < 1; i += 1) {
    switch (i) {
    case 1:
        continue;
        break;
    default:
        break;
    }
}
```

This time, I get another error:

``` bash
test.js
  6:8  error  Found unexpected statement after a continue  no-unreachable
```

Although I can work around this by using a `//falls through` comment instead of the `break;`, the code is technically never falling through and the comment will be misleading in that case.
